contracts/Remittance.sol

```diff
@@ -1,38 +1,40 @@
 pragma solidity >=0.4.21 <0.6.0;
 
 import "./Pausable.sol";
-import "./Ownable.sol";
 
 contract Remittance is Pausable {
-  address public carol;
-  bytes32 private withdrawHash;
+  struct Escrow {
+    address carol;
+    uint256 amount;
+  }
+
+  mapping (bytes32 => Escrow) public escrows;
 
   event LogDeposited(address indexed sender, uint256 amount, bytes32 hash, address indexed carol);
-  event LogWithdrawn(address indexed sender, uint256 amount);
+  event LogWithdrawn(address indexed sender, bytes32 hash, uint256 amount);
 
   constructor(bool paused) Pausable(paused) public {
   }
 
-  function generateHash(address _carol, bytes32 _bobPassword, bytes32 _carolPassword) public view returns(bytes32) {
-     return keccak256(abi.encodePacked(this, _carol, _bobPassword, _carolPassword));
+  function generateHash(address carol, bytes32 bobPassword, bytes32 carolPassword) public view returns(bytes32) {
+     return keccak256(abi.encodePacked(this, carol, bobPassword, carolPassword));
   }
 
-  function deposit(bytes32 _hash, address _carol) external payable onlyOwner whenRunning whenAlive {
-    require(_carol != address(0), "Carol's address must not be zero!");
+  function deposit(bytes32 hash, address carol) external payable onlyOwner whenRunning whenAlive {
```

Comment:
Now you need to open the `deposit` facility to anyone. Not just the owner. Anyone can deposit with any hash.